### PR TITLE
 #157 Add quick actions on email

### LINKED ISSUE #157: Add quick actions on emails
On browser to delete an email in the email list view, the user has to select the email and then click on the delete button on the top right.

## Proposal

This can be enhanced to save time and clicks by adding actions on emails "à la gmail" where actions are displayed when the mouse is over an email:

Capture d’écran 2020-09-18 à 09 38 45

### PR DESCRIPTION
**Pull request by MichaelBailly**
- trash
- change read/unread status

Resolves #157

inbox-hover

### CODE REVIEW

**Review by rezk2ll — DISMISSED** [review]

**Comment by chamerling** [comment]
I am sad that my face is not on your desktop :p

**Review by chamerling — APPROVED** [review]

**Comment by fabienmoyon** [comment]
On medium device you displayed icons on the clock and not in XL device, is it desired ?

**Comment by MichaelBailly** [comment]
> On medium device you displayed icons on the clock and not in XL device, is it desired ?

I porcessed like we decided during our mmocking phase 2 days ago. I checked just after our discussion today and indeed the attachment icon appears at the left of the star, which make it not desirable to put the trash icon there (because the trash icon would not be always at the same place in the widget, depending of whether the attachment icon is displayed or not).

image

File: src/linagora.esn.unifiedinbox/app/directives/lists.spec.js [hunk 1]
```diff
@@ -449,6 +449,7 @@ describe('The linagora.esn.unifiedinbox List module directives', function() {
         $scope.item = {
           moveToMailboxWithRole: sinon.spy(function() {return $q.when();}),
           isUnread: true,
+          mailboxIds: [],
```
Comment (tuanlc) [inline src/linagora.esn.unifiedinbox/app/directives/lists.spec.js:452]: I don't see any role in this PR. Can you explain it?

**Review by tuanlc — COMMENTED** [review]
Cool improvement!

File: src/linagora.esn.unifiedinbox/views/email/list/list-item.pug [hunk 2]
```diff
@@ -30,6 +30,10 @@
             inbox-emailer(ng-if="mailboxRole.value !== 'sent'", emailer="item.from")
             inbox-emailer-group(ng-if="mailboxRole.value === 'sent' && item.emailRecipients", group="item.emailRecipients", display-inline="true")
         .inbox-list-item-date-and-indicators
+          .inbox-hover-actions.flex-row(ng-click='$event.stopPropagation(); $event.preventDefault()')
+              i.mdi.mdi-delete.waves-notransition(ng-if='ctrl.canTrashMessages()' style='' ng-click='ctrl.moveToTrash()' title="{{'Delete' | translate}}")
+              i.mdi.mdi-eye.waves-notransition(ng-if='item.isUnread' style='' ng-click='ctrl.markAsRead()' title="{{'Mark as read' | translate}}")
+              i.mdi.mdi-eye-off.waves-notransition(ng-if='!item.isUnread' style='' ng-click='ctrl.markAsUnread()' title="{{'Mark as unread' | translate}}")
```
Comment (huy-ta) [inline src/linagora.esn.unifiedinbox/views/email/list/list-item.pug:36]:
```suggestion
              i.mdi.mdi-delete.waves-notransition(ng-if='ctrl.canTrashMessages()', ng-click='ctrl.moveToTrash()', title="{{ 'Delete' | translate }}")
              i.mdi.mdi-eye.waves-notransition(ng-if='item.isUnread', ng-click='ctrl.markAsRead()', title="{{ 'Mark as read' | translate }}")
              i.mdi.mdi-eye-off.waves-notransition(ng-if='!item.isUnread', ng-click='ctrl.markAsUnread()', title="{{ 'Mark as unread' | translate }}")
```

Btw, is there any reason why you must explicitly specify `style=''` here?

File: src/linagora.esn.unifiedinbox/app/directives/lists.spec.js (hunk 1 above)
Comment (MichaelBailly) [reply src/linagora.esn.unifiedinbox/app/directives/lists.spec.js:452]: The `ctrl.canTrashMessages()` call in the `src/linagora.esn.unifiedinbox/views/email/list/list-item.pug` template do calls to an inner function that requires this property on the item object.

File: src/linagora.esn.unifiedinbox/views/email/list/list-item.pug (hunk 2 above)
Comment (MichaelBailly) [reply src/linagora.esn.unifiedinbox/views/email/list/list-item.pug:36]: no reason, you're right. That's garbage from my early code. Thanks

File: src/linagora.esn.unifiedinbox/app/directives/lists.spec.js (hunk 1 above)
Comment (tuanlc) [reply src/linagora.esn.unifiedinbox/app/directives/lists.spec.js:452]: Thank for the explanation!

**Review by tuanlc — APPROVED** [review]

**Review by huy-ta — APPROVED** [review]

**Review by rezk2ll — APPROVED** [review]

File: src/linagora.esn.unifiedinbox/css/modules/listItems.less [hunk 3]
```diff
@@ -279,6 +284,26 @@
           }
         }
       }
+
+      .inbox-hover-actions {
+        // in esn-frontend-common-libs/src/frontend/components/material-admin/less/inc/card.less
+        // color is hardcoded
+        background-color: #fff;
```
[the remark this answers is not in the document]
Comment (fabienmoyon) [reply src/linagora.esn.unifiedinbox/css/modules/listItems.less:291]: fu****king pending review..

[the remark this answers is not in the document]
Comment (fabienmoyon) [reply src/linagora.esn.unifiedinbox/css/modules/listItems.less:291]: themes colors : esn-frontend-common-libs/src/frontend/css/colors.less

inbox colors : src/linagora.esn.unifiedinbox/css/modules/listItems.less
& src/linagora.esn.unifiedinbox/css/styles.less

Need to be tested

[the remark this answers is not in the document]
Comment (fabienmoyon) [reply src/linagora.esn.unifiedinbox/css/modules/listItems.less:291]: Reminder : Need to fix hover background color on smaller and middle device and need to hide actions on dragging an email

Capture d’écran de 2020-09-25 14-59-17
Capture d’écran de 2020-09-25 14-58-56

Capture d’écran de 2020-09-25 14-59-22

**Pull request merged**